File: app/src/main/java/com/example/despensacx/TiendaDao.java
package com.example.despensacx;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface TiendaDao {
    @Insert
    long insert(TiendaEntity tienda);

    @Update
    void update(TiendaEntity tienda);

    @Delete
    void delete(TiendaEntity tienda);

    @Query("SELECT * FROM tiendas ORDER BY orden ASC, nombre ASC")
    LiveData<List<TiendaEntity>> getAllTiendas();

    @Query("SELECT * FROM tiendas ORDER BY orden ASC, nombre ASC")
    List<TiendaEntity> getAllTiendasSync();

    @Query("SELECT COUNT(*) FROM tiendas")
    int getCountSync();

    @Query("SELECT COUNT(*) FROM productos WHERE tiendaId = :tiendaId")
    int countProductosByTiendaSync(long tiendaId);
}

File: app/src/main/kotlin/com/example/despensacx/data/TiendaDao.kt
package com.example.despensacx.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface TiendaDao {
    @Insert
    suspend fun insert(tienda: TiendaEntity): Long

    @Update
    suspend fun update(tienda: TiendaEntity)

    @Delete
    suspend fun delete(tienda: TiendaEntity)

    @Query("SELECT * FROM tiendas ORDER BY orden ASC, nombre ASC")
    fun getAllTiendas(): LiveData<List<TiendaEntity>>

    @Query("SELECT * FROM tiendas ORDER BY orden ASC, nombre ASC")
    fun getAllTiendasSync(): List<TiendaEntity>

    @Query("SELECT COUNT(*) FROM tiendas")
    fun getCountSync(): Int

    @Query("SELECT COUNT(*) FROM productos WHERE tiendaId = :tiendaId")
    fun countProductosByTiendaSync(tiendaId: Long): Int

    @Query("DELETE FROM tiendas")
    suspend fun deleteAll()
}
